code/game/objects/objs.dm

```diff
@@ -364,3 +364,20 @@
 	for(var/reagent in reagents)
 		var/datum/reagent/R = reagent
 		. |= R.expose_obj(src, reagents[R])
+/*
+only drops anything if the src doesnt have the NODECONSTRUCT_1 or HOLOGRAM_1 flags
+*/
+/obj/proc/destroy_drop(var/obj/destroyed, var/obj/to_drop = null, var/amount = 1)
```

Comment:
What kind of niche cases would require `destroyed` to be a different object than `src`?

I don't like how this bundles so much behavior together, basically makes use of what amounts to `istype(src)` checks, and seems to simply replace `NODECONSTRUCT_1` flag checks even when there's no intent to use the rest of the behavior.

A better way of doing this would be for the procs to call parent to process the deconstruction, and for checks like this to be present at the base of the proc.